app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -153,6 +192,76 @@ class UserViewModel(
     }
   }
 
+  /**
+   * Upload the picture to the cloud storage update the user object in the database with the cloud
+   * path created. update the user object in the viewmodel with the url of the image and the cloud
+   * path
+   *
+   * @param context the context used to resolve the image uri
+   */
+  private fun uploadProfilePicture(context: Context) {
+    if (imageRepository == null) {
+      Log.e("UserViewModel", "ImageRepository is null, should not upload image")
+      return
+    }
+    val user = currentUser.value ?: return
+    val fileUri = user.localSession?.profilePictureUri ?: return
+    val destinationPath = "profile_pictures/${user.public.userId}"
+    imageRepository.uploadImage(
+        context = context,
+        fileUri = fileUri,
+        destinationPath = destinationPath,
+        onSuccess = { url ->
+          Log.d("UserViewModel", "Image uploaded successfully to $url")
+          val updatedUser =
+              user.copy(
```

Comment:
why do you copy the User, then use the update function? (I know it's functionally identical to addUser, but the way you've implemented it as is a little confusing...)

Reply:
Not sure I understand

User is immutable if I want to modify some fields I need to do a .copy, here I want to update the user with the new Url I just got, so I do a copy then I call updateUser so that it replaces the user document and the user in the viewmodel with this new instance

This is the same mechanism as in the edit profile screen.